### PR TITLE
[style-value-parser] process disjoint, intersecting, and subset ranges in media queries

### PR DESCRIPTION
**Pull request by mellyeliu**
Updated mergeIntervalsForAnd to properly merge and simplify ranges for min-width / max-width pairs:
- build numeric intervals for each dimension (width/height). it collects all the lower and upper bounds from the rules and determines the tightest range:
   - for min-width it takes the largest lower bound
   - for max-width it takes the smallest upper bound

Added handling for the special case of overlapping queries inside an outer range. If there's a gap in the valid intervals, the function now splits the outer range into valid disjoint intervals on either side of the hole.
 - if the hole fully overlaps the lower or upper side of the outer range, we drop the empty side
 - if both sides are valid, we return them as an or group so the final query represents both disjoint intervals
 - if only one side is valid, we just return that branch directly instead of wrapping it in or

Added tests for all combinations of ranges:
- disjoint ranges (completely separate, media query should be preserved)
- overlapping ranges (just decrease the bounds of the parent interval)
- subset ranges (split into two ranges, a, c becomes a, b and b, c)

### CODE REVIEW

File: packages/style-value-parser/src/at-queries/media-query.js [hunk 1]
```diff
@@ -514,14 +539,23 @@ export class MediaQuery {
           : `(not ${this.#toString(queries.rule)})`;
       case 'and':
         return queries.rules.map((rule) => this.#toString(rule)).join(' and ');
-      case 'or':
+      case 'or': {
+        const validRules = queries.rules.filter(
```
Comment (mellyeliu) [inline packages/style-value-parser/src/at-queries/media-query.js:543]: ideally this edge case should be handled in the `normalize` function, but I had difficulty structuring the recursion in a way that handled all edge cases - open to improvements

File: packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js [hunk 2]
```diff
@@ -418,7 +418,59 @@ describe('Media Query Transformer', () => {
...
+        gridColumn: {
+          default: '1 / 2',
+          '@media (max-width: 1440px) and (min-width: 900px)': '1 / 4',
+          '@media (max-width: 800px) and (min-width: 600px)': '1 / 3',
+          '@media (max-width: 500px) and (min-width: 400px)': '1 / 1',
+        },
+      },
+    };
+
+    const expectedStyles = {
```
Comment (mellyeliu) [inline packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js:458]: for disjoint ranges we expect the output to be the same (as the developer has already ensured they do not overlap)

File: packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js [hunk 3]
```diff
@@ -443,4 +494,87 @@ describe('Media Query Transformer', () => {
...
+          '@media (max-width: 1050px) and (min-width: 1010px)': '1 / -1',
+        },
+      },
+    };
+
+    const expectedStyles = {
+      foo: {
+        gridColumn: {
+          default: '1 / 2',
+          '@media (min-width: 900px) and (max-width: 999.99px) or (min-width: 1100.01px) and (max-width: 1440px)':
```
Comment (mellyeliu) [inline packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js]: these ranges are narrowed to whatever new min/max restriction from the queries below from the intersecting parts

File: packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js [hunk 4]
```diff
@@ -443,4 +494,87 @@ describe('Media Query Transformer', () => {
...
+          '@media (max-width: 1100px) and (min-width: 1000px)': '1 / 3',
+        },
+      },
+    };
+
+    const expectedStyles = {
+      foo: {
+        gridColumn: {
+          default: '1 / 2',
+          '@media (min-width: 900px) and (max-width: 999.99px) or (min-width: 1100.01px) and (max-width: 1440px)':
```
Comment (mellyeliu) [inline packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js]: these are now split into two ranges because the below query creates a "hole"

File: packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js [hunk 5]
```diff
@@ -443,4 +494,87 @@ describe('Media Query Transformer', () => {
...
+          '@media (max-width: 500px) and (min-width: 400px)': '1 / 1',
+        },
+      },
+    };
+
+    const expectedStyles = {
+      foo: {
+        gridColumn: {
+          default: '1 / 2',
+          '@media (min-width: 900px) and (max-width: 999.99px) or (min-width: 1100.01px) and (max-width: 1440px)':
```
Comment (nmn) [inline packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js]:
```suggestion
          '@media ((min-width: 900px) and (max-width: 999.99px)) or ((min-width: 1100.01px) and (max-width: 1440px))':
```

The toString part of this has a bug.

File: packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js (hunk 4 above)
Comment (nmn) [reply packages/style-value-parser/src/at-queries/__tests__/media-query-transform-test.js]: Same, needs brackets around the "and" ranges.

**Review by nmn — CHANGES_REQUESTED** [review]
This is nearly ready. The simplification logic seems to be working well.

All that remains is the one small bug fix when stringifying disjoint ranges.

**Comment by mellyeliu** [comment]
fixed parenthesis bug in ranges in `toString`

**Review by vincentriemer — APPROVED** [review]
